HPXMLtoOpenStudio/resources/constructions.rb

```diff
@@ -1813,6 +1813,29 @@ def self.check_surface_assembly_rvalue(runner, surfaces, inside_film, outside_fi
       end
     end
   end
+
+  def self.get_ufactor_shgc_adjusted_by_storms(storm_type, base_ufactor, base_shgc)
+    # Ref: https://labhomes.pnnl.gov/documents/PNNL_24444_Thermal_and_Optical_Properties_Low-E_Storm_Windows_Panels.pdf
+    # U-factor and SHGC adjustment based on the data obtained from the above reference
+    if base_ufactor < 0.45
+      fail "Invalid base window U-Factor for storm windows upgrade '#{base_ufactor}'"
+    end
```

Comment:
We need to call this out in the documentation so that a user/developer is not surprised by this error.

Reply:
And actually, this error message should be revised. In HEScore, storm windows only ever show up for an "upgrade", but that's not true for OS-HPXML. How about:

```suggestion
    if base_ufactor < 0.45
      fail "Unexpected base window U-Factor (#{base_ufactor}) for a storm window."
    end
```

Reply:
And you should really add a test to the `test_measure_error_messages` section of `test_validation.rb`.